### PR TITLE
[MAINTENANCE] Test serialization of text, table, and bulleted list `rendered_content` in `ExpectationValidationResult`

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-911/GREAT-962
- GREAT-911/GREAT-964
- GREAT-911/GREAT-965

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Comment by ghost** [comment]
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=5438&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=5438&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=5438&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map Legend

File: tests/render/test_inline_renderer.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+
+@pytest.mark.parametrize(
+    "expectation_configuration,expected_serialized_expectation_configuration_rendered_atomic_content,expected_serialized_expectation_validation_result_rendered_atomic_content",
+    [
+        pytest.param(
+            ExpectationConfiguration(
+                expectation_type="expect_table_row_count_to_equal",
+                kwargs={"value": 3},
+            ),
```
Comment (cdkini) [inline tests/render/test_inline_renderer.py:24]: Could we give each of these params an id or name so we're able to access them with `pytest -k`?

**Review by cdkini — APPROVED** [review]

File: tests/render/test_inline_renderer.py (hunk 1 above)
Comment (anthonyburdi) [reply tests/render/test_inline_renderer.py:24]: I was about to write the same thing 🙇

Comment (NathanFarmer) [reply tests/render/test_inline_renderer.py:24]: I added an id to each of the test cases and confirmed it works with `pytest -k` ✅

**Pull request merged**